base/src/main/java/org/mozilla/jss/ssl/javax/JSSSession.java

```diff
@@ -149,8 +158,12 @@ public boolean isValid() {
 
     @Override
     public void invalidate() {
-        if (parent.getSSLFDProxy() != null) {
-             SSL.InvalidateSession(parent.getSSLFDProxy());
+        JSSEngine jssEngine = parent.get();
+        if (jssEngine == null) {
+          return;
+        }
+        if (jssEngine.getSSLFDProxy() != null) {
+             SSL.InvalidateSession(jssEngine.getSSLFDProxy());
```

Comment:
The return value of `jssEngine.getSSLFDProxy()` probably should be stored in a variable so it can be reused.

Reply:
Saved as ssl_fd.  I would have used sslfdProxy, but the code already uses ssl_fd everywhere else.